### PR TITLE
Auto screenshot bug fix

### PR DESCRIPTION
**Pull request by pmpranav**
Adding a try catch block so that the screenshot doesn't fail the test case when encountering killed webdriver sessions as reporter here, https://github.com/paypal/nemo-screenshot/issues/21

### CODE REVIEW

**Comment by ibihim** [comment]
Wouldn't it be easier to read if the try-catch would be placed into a reused and complexity hiding function, e.g. "safeSnap"?

File: index.js [hunk 1]
```diff
@@ -150,8 +154,12 @@ module.exports = {
       nemo.driver.flow_.on('uncaughtException',function(exception){
         var filename = 'ScreenShot_onException-' + process.pid + '-' + new Date().getTime();
         var screenShotFileName = path.resolve(screenShotPath, filename);
-        nemo.screenshot.snap(screenShotFileName).then(function(){
-        });
+        try{
+          nemo.screenshot.snap(screenShotFileName).then(function(){
```
Comment (ibihim) [inline index.js]: Why is there a follow up then, with an empty function?

Comment (pmpranav) [inline index.js]: You are right. This function is unnecessary. Will remove it when I update the pull request.

**Comment by pmpranav** [comment]
@ibihim Is it just a safesnap function used inside the module or a nemo.screenshot.safeSnap that is also made part of the public API. Just to clarify

**Comment by ibihim** [comment]
@pmpranav I am referring to a function inside of the module, which is not part of the API. Maybe there are better names than "safeSnap". 
Robert C. Martin does recommend that in his book Clean Code (Chapter 3 - Extract Try/Catch Blocks). It makes the code cleaner and easier to read. 

This is just a suggestion :)

**Comment by pmpranav** [comment]
Updated pull request to address the suggestions provided by @ibihim

**Comment by grawk** [comment]
I will take this PR code and test it locally to see how it works and if there are any recommended improvements.

**Comment by grawk** [comment]
I've opened https://github.com/paypal/nemo-screenshot/pull/28 for this issue. It gets to the heart of the issue, which is that sometimes (for whatever reason) the exception event handler gets triggered for a dead session. I've also resolved a couple other issues I ran into. Most importantly, adding the exception handler "once" and throwing the exception in the promise resolution block.

**Pull request closed without merging**